src/Router.php

```diff
@@ -408,11 +408,30 @@ public static function get_raw_data() {
 	 * This processes the graphql requests that come into the /graphql endpoint via an HTTP request
 	 *
 	 * @since  0.0.1
+	 * @global WP_User $current_user The currently authenticated user.
 	 * @return mixed
 	 * @throws \Exception Throws Exception.
 	 * @throws \Throwable Throws Exception.
 	 */
 	public static function process_http_request() {
+		/* @var WP_User|null $current_user */
+		global $current_user;
```

Comment:
@therealgilles I like this because it only affects GraphQL HTTP Requests, but won't affect GraphQL operations that are done within PHP (such as https://www.wpgraphql.com/2017/09/20/using-graphql-queries-in-php-within-your-wordpress-theme-or-plugin/)

Reply:
The REST API server_request code was added recently to WP when they integrated Application Passwords in version 5.6. The documented reasoning is a bit different than the problem I encountered. This may actually solve two problems at once: (1) It ensures authentication is done correctly once everything is setup, or (2) re-done in my case when it could not be done properly earlier. A tiny bit surprised the WP developers did not create some kind of _wp_reset_current_user_if_zero function.